devsearch/tree_sitter/parsing.py

```diff
@@ -0,0 +1,38 @@
+from functools import lru_cache
+import logging
+from typing import Iterator
+
+from devsearch.tree_sitter.setup import parser_library
+
+logger = logging.getLogger(__name__)
+
+
+@lru_cache(maxsize=None)

```

Comment:
the same suggestion as above - explicit dictionary to store results instead of `lru_cache`